### PR TITLE
Redesign review component

### PR DESCRIPTION
**Pull request by jimchamp**
Redesigns the books page stats component, making it somewhat less of a wall of text.

This will need some tweaking, and it will be good to get feedback about the same.

### Testing

Renamed `stats_component.html` to better reflect what the component does.  Removed unused parameters from the component.

Did not wrap label and reviews of a single category in one HTML element, as this was causing undesired line wrapping.  The same is true for the label itself and all values of a single type.

### Screenshot

no_review_logged_in
_No reviews, patron logged in_

no_review_logged_out
_No reviews, patron logged out_

review_logged_in
_Reviews exist, patron logged in_

review_logged_out
_Reviews exist, patron logged out_

### Stakeholders

@mekarpeles 
@cdrini

### CODE REVIEW

**Comment by mekarpeles** [comment]
This looks **significantly** better than before, great job @jimchamp. I'd be content with this going out as is

Nitpicks:
- Whether you're logged in / logged out, the `+ [Log in to] Add your community reviews` should likely be in the same consistent left-aligned position w/ no `<hr>`
- We use border-radius 5px elsewhere on our carousels; I wonder if we want to keep a consistent degree for the rounded aesthetic
- Opinions on a lighter color / non-bold font-weight for the counts portion? e.g. `(2)`
- Opinions on decreasing font size?

**Comment by mekarpeles** [comment]
Other enhancements which seem worth bringing up in the future but reasonably out of scope for now:
- Some sort of very minor color coding hint (e.g. a bullet) for the. One idea is to make the left side of the labels not have border-radius and instead be a colored 2px line?
- Comparing a version which always begins the line with category name

**Review by mekarpeles — APPROVED** [review]
lgtm, some useful feedback offered which may be addressed in a followup if deemed necessary

**Comment by jimchamp** [comment]
A 15px font size looks alright.  Border radius of 5px makes things look pretty blocky:

Screenshot from 2021-08-17 05-17-54

Agree with points about count font weight and the login link.

File: openlibrary/templates/observations/review_component.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<a name="reader-observations" class="section-anchor"></a>      
+
+<div class="tab-section review-component">
+  <h2 class="observation-title">$_('Community Reviews') ($total_reviews)</h2>
+  $if total_reviews > 0:
+    <div class="reviews">
+    $for o in reader_observations['observations']:
+      $ total = o['total_respondents_for_type']
+        <span class="label" title="$o['description']">$o['label'] ($total)</span>
```
Comment (cdrini) [inline openlibrary/templates/observations/review_component.html:16]: This classname is too generic; it's conflicting with other things on the page. Let's use BEM:

```html
        <span class="reviews__label" title="$o['description']">$o['label'] ($total)</span>
```

File: openlibrary/templates/observations/review_component.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          $ percentage = (v['count'] / total) * 100
+          <span class="pill">
+            <span class="value">
+              $v['value']
+            </span>
+            <span class="percentage">
+              $int(percentage)%
+            </span>
+          </span>
+        <span class="margin"></span>
```
Comment (cdrini) [inline openlibrary/templates/observations/review_component.html:27]: Why is this needed?

File: openlibrary/templates/observations/review_component.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    $for o in reader_observations['observations']:
+      $ total = o['total_respondents_for_type']
+        <span class="label" title="$o['description']">$o['label'] ($total)</span>
+        $for v in o['values']:
+          $ percentage = (v['count'] / total) * 100
+          <span class="pill">
+            <span class="value">
+              $v['value']
+            </span>
+            <span class="percentage">
```
Comment (cdrini) [inline openlibrary/templates/observations/review_component.html:23]: Ditto with all these classes; just prefix them with `reviews__`.

File: openlibrary/templates/observations/review_component.html (hunk 1 above)
Comment (cdrini) [reply openlibrary/templates/observations/review_component.html:16]: Can we place these all in a wrapper span? That'll make some styling possible.

Eg

```html
<span class="reviews__group">
    <span class="reviews__label">...</span>
    $for ...:
        <span class="reviews__pill">...</span>
```

Comment (cdrini) [reply openlibrary/templates/observations/review_component.html:16]: Something like this would be nice; that'll let us style this as well. You can add the parenthesis using CSS.

```html
        <span class="label" title="$o['description']">$o['label'] <span class="reviews__numeric">$total</span></span>
```

**Review by cdrini — COMMENTED** [review]
1 blocker; the CSS is leaking and affecting some other elements on the page. And a few nice to have to make this easier to style :)

File: static/css/components/observationStats.less [hunk 4]
```diff
@@ -1,25 +1,67 @@
...
 }
 
-.observation-header {
-  font-size: 14px;
-  margin: 0 0 1em;
+.reviews {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  margin-bottom: .5em;
```
Comment (cdrini) [inline static/css/components/observationStats.less:16]: Not a blocker; I would recommend using px; ems or _very_ confusing to deal with. I haven't looked too much into it, but anyone can understand and use pixels, whereas `em`s are very hard to reason about :P

File: openlibrary/templates/observations/review_component.html (hunk 1 above)
Comment (jimchamp) [reply openlibrary/templates/observations/review_component.html:16]: > Can we place these all in a wrapper span? That'll make some styling possible.
Will likely need some help getting everything to wrap properly once this change is made.
